### DISCUSSION TITLE
Is this tool works for you?

### DISCUSSION THREAD

**Discussion by KnugiHK** [KnugiHK/WhatsApp-Chat-Exporter#17] [Polls]
As per discussion in issue https://github.com/KnugiHK/Whatsapp-Chat-Exporter/issues/9, WhatsApp may adopt "message" or "messages" table to manage its conversations. I want to know if this tool works for you and how many users are affected by the issue mentioned above.

After you have run the program, I would like to invite you to comment below and state if this tool works, what are the WhatsApp versions (both in account creation and now) and the Android version that you are working on. Thank you!

**Comment by monitorebr** [comment]
estou tentando de todas as formas exportar um backup que tenho no pc, pois nao tenho mais o aparelho... entao nao consigo resgatar a chave apartir das conversas
alguem tem alguma dica para me passar?

**Reply by KnugiHK** [reply]
You could try to log in into WhatsApp in another device then extract the key from that device.

PS: I used the translation provided by GitHub, I am not sure if I understand you correctly.

**Reply by monitorebr** [reply]
seria possivel, 
mas o numero foi ativado para outro usuario...
e estou precisando das mensagens contidas no meu backup, pois tem conversas de trabalho muito importantes...

**Reply by KnugiHK** [reply]
1. Did you enable end-to-end encryption on the backup (the crypt15)? If yes, decrypt your backup with the 64 digits encryption key which was shown when you enabled crypt15 backup.

2. If you don't have crypt15 backup or do not have access to the 64 digits encryption key, did you enable cloud backup? Google Drive backup can be retrieved through some hack, but for iCloud, I am not sure if it can be retrieved.

**Reply by monitorebr** [reply]
o backup ainda é crypt14

vamos ver se entendi.

criar em um novo dispositivo um backup em crypt15 com um numero novo...
acrescentar na pasta do whatsapp os arquivos que tenho,
criar um novo backup com crypt15
extrair a chave do novo backup
tentar extrair as conversas direto do aparelho com a nova chave crypt15


seria este procedimento?

grato pelo apoio

**Reply by KnugiHK** [reply]
If your backup is still crypt14, you should register the same number in another device that you control, then extract the key with [WhatsApp-Key-DB-Extractor](https://github.com/KnugiHK/WhatsApp-Key-DB-Extractor).

If you cannot get the number back, and you have cloud backup, you can retrieve it from cloud.

If you do not have the number, cloud backup or the original device, AFAIK there is nothing you can do to decrypt the backup in your PC.

**Reply by monitorebr** [reply]
ola amigo

tenho apenas a copia da pasta whatsapp completa, nao existe mais o numero

**Reply by Leegray28** [reply]
Hi KnugiHK,

Really looking for your help with this. I'm about to lose 10+ years of WhatsApp backup and I'm devastated. 

Any help you can offer would be very much appreciated. 

Kind regards,

Lee

**Reply by KnugiHK** [reply]
Hi @Leegray28 
Feel free to poke me at [matrix](https://matrix.to/#/#wtsexporter:matrix.org).

**Reply by thanhnguyenretailcircle** [reply]
Hi @KnugiHK 

We're currently looking WhatsApp exporting function and found your tool really promising. We would like to collaborate with you for a custom solution if you are available?

Is there anyway I can contact you directly?

Thank you and looking forward to hear back from you.

Kind regards,
T

**Reply by KnugiHK** [reply]
> Hi @KnugiHK
> 
> We're currently looking WhatsApp exporting function and found your tool really promising. We would like to collaborate with you for a custom solution if you are available?
> 
> Is there anyway I can contact you directly?
> 
> Thank you and looking forward to hear back from you.
> 
> Kind regards, T

Hi @thanhnguyenretailcircle,

You may poke me at [matrix](https://matrix.to/#/#wtsexporter:matrix.org) or send me an email. For email, in case of anything confidential, you can find my PGP key at https://keyserver.pgp.com.

**Comment by ghost** [comment]


**Reply by KnugiHK** [reply]
Hi. If you believe the situation you mentioned is a bug, please open an issue for better tracking.

As for the feature you requested, it is tracked in #80.
